### PR TITLE
Run CI Jobs via Github Actions

### PR DESCRIPTION
**Pull request by joshua-goldstein**
This PR will add workflows to run Ristretto tests.  It also does some basic cleanup and updates the code owners.  

Notable points:
- go version is now pinned in .go-version and referenced by the workflow
- test.sh is deprecated since it only was only relevant for TeamCity environments (it was a fancy wrapper on top of `go test -v ./...`)
- ci-ristretto-tests and ci-ristretto-lint bring this repository to parity with the dgraph repository
- main.yml was removed since we are no longer automatically closing issues
- Coveralls now used for test coverage
- Issue template updated
- Pull Request template added

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # CODEOWNERS info: https://help.github.com/en/articles/about-code-owners
 # Owners are automatically requested for review for PRs that changes code
 # that they own.
-* @manishrjain @martinmr @jarifibrahim
+* @akon-dey @nosql22
```
Comment (skrdgraph) [inline .github/CODEOWNERS:4]: I would suggest you to add your user and me to this as well.

File: .github/workflows/ci-ristretto-tests.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Run Unit Tests
+        run: go test -race -covermode atomic -coverprofile=covprofile ./...
+      - name: Install Goveralls
+        run: go install github.com/mattn/goveralls@latest
+#      - name: Send Coverage Results
+#        env:
+#          COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+#        run: goveralls -coverprofile=covprofile -service=github
+#      - name: Send Coverage Results
+#        run: cd t && goveralls -repotoken ${{ secrets.COVERALLSIO_TOKEN }} -coverprofile=coverage.out
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-tests.yml:35]: can you open another PR with these uncommented? once the access issue is resolved, we will get the coverage also setup.

File: .github/workflows/ci-ristretto-lint.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: ci-ristretto-lint
+on:
+  push:
+    branches:
+      - main
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-lint.yml:5]: change to `master`

File: .github/workflows/ci-ristretto-lint.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request:
+    branches:
+      - main
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-lint.yml:8]: change to `master`

File: .github/workflows/ci-ristretto-tests.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+name: ci-ristretto-tests
+on:
+  push:
+    branches:
+      - main
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-tests.yml:5]: change to `master`

File: .github/workflows/ci-ristretto-tests.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+name: ci-ristretto-tests
+on:
+  push:
+    branches:
+      - main 
+  pull_request:
+    branches:
+      - main
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-tests.yml:8]: change to `master`

**Review by skrdgraph — COMMENTED** [review]
I will get the access hopefully tomorrow.

File: .github/CODEOWNERS (hunk 1 above)
Comment (joshua-goldstein) [reply .github/CODEOWNERS]: Done

File: .github/workflows/ci-ristretto-lint.yml (hunk 3 above)
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-lint.yml:5]: ^ ignore, will get addressed once the access issue is resolved

File: .github/workflows/ci-ristretto-lint.yml (hunk 4 above)
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-lint.yml:8]: ^ ignore, will get addressed once the access issue is resolved

File: .github/workflows/ci-ristretto-tests.yml (hunk 5 above)
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-tests.yml:5]: ^ ignore, will get addressed once the access issue is resolved

File: .github/workflows/ci-ristretto-tests.yml (hunk 6 above)
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-tests.yml:8]: ^ ignore, will get addressed once the access issue is resolved

File: .github/workflows/ci-ristretto-tests.yml (hunk 2 above)
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-tests.yml:35]: ^ ignore, will get addressed once the access issue is resolved

File: .github/workflows/ci-ristretto-lint.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  go-lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golang-lint
+        env:
+          # prevent OOM
+          GOGC: 10
+        uses: golangci/golangci-lint-action@v2
```
Comment (skrdgraph) [inline .github/workflows/ci-ristretto-lint.yml:21]: to control what linters run, you will need to add something like this https://github.com/dgraph-io/dgraph/blob/main/.golangci.yml

**Review by skrdgraph — COMMENTED** [review]
need to add linter yaml file

File: .github/workflows/ci-ristretto-lint.yml (hunk 7 above)
Comment (joshua-goldstein) [reply .github/workflows/ci-ristretto-lint.yml:21]: Done (only running two lint tests now as in dgraph but we can open it up later)

**Review by skrdgraph — APPROVED** [review]

File: .github/workflows/ci-ristretto-tests.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+          #!/bin/bash
+          DEFAULT_VERSION="1.17"
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; } || echo $DEFAULT_VERSION)
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Run Unit Tests
+        run: go test -race -covermode atomic -coverprofile=covprofile ./...
```
[the remark this answers is not in the document]
Comment (joshua-goldstein) [reply .github/workflows/ci-ristretto-tests.yml:27]: I took that from the previous test.sh script.  Probably it is a good idea to keep it around (only downside is tests may take longer).

[the remark this answers is not in the document]
Comment (skrdgraph) [reply .github/workflows/ci-ristretto-tests.yml:27]: 👍

**Pull request merged**